test/CollectorTest.php

```diff
@@ -139,4 +139,135 @@ private function copy($sPattern)
 			}
 		}
 	}
+	
+	/**
+	 * @dataProvider providerUpdateSDSAttributes
+	 * @param array $aExpectedAttrDef
+	 * @param array $aSynchroAttrDef
+	 * @param bool $bWillUpdate
+	 */
+	public function testUpdateSDSAttributes($aExpectedAttrDef, $aSynchroAttrDef, bool $bWillUpdate)
+	{
+	    $this->copy(APPROOT."/test/collector/attribute_isnullified/*");
+	    require_once APPROOT."/core/restclient.class.inc.php";
+	    require_once self::$sCollectorPath."iTopPersonCollector.class.inc.php";
+	    $oCollector = new iTopPersonCollector();
+	    $oMockClient = $this->CreateMock('RestClient');
+	    $oMockClient->expects($this->exactly($bWillUpdate ? 1 : 0))->method("Update")->willReturn(['code' => 0]);
```

Comment:
maybe you could add with() with proper parameters passed to Update function? more for documentation purpose.